src/classes/VOL_CTRL_PersonalSiteContactInfo.cls

```diff
@@ -279,26 +279,30 @@ global with sharing class VOL_CTRL_PersonalSiteContactInfo {
         return null;
     }
 
-    global PageReference save() { 
-        if (contactReadOnly != null) {          
-            Set<String> contactFieldsToUpdate = new Set<String>();
+    global PageReference save() {
+        try {
+            if (contactReadOnly != null) {
+                Set<String> contactFieldsToUpdate = new Set<String>();
+
+                //  now copy over all the non-null fields from the form's contact to the existing contact.
+                for (string strF : listStrContactFields) {
+                    if (contactEdit.get(strF) != null) {
+                        try {
```

Comment:
Is this try/catch block needed? If so, why is the code swallowing/ignoring the exception?

Reply:
// we need this catch because some of the fields in listStrContactFields
                            // are readonly formula fields.  We can't use describes for isCreatable,
                            // because we won't know the difference between a formula field vs. one
                            // they don't have permissions to (which we will check below).